glean-core/ios/Glean/Metrics/JweMetric.swift

```diff
@@ -0,0 +1,179 @@
+    }
+
+    /// Build a JWE value from it's elements and set to it.
+    ///
+    /// - parameters:
+    ///     * header: A variable-size JWE protected header.
+    ///     * key: A variable-size encrypted key.
+    ///            This can be an empty octet sequence.
+    ///     * initVector: A fixed-size, 96-bit, base64 encoded Jwe initialization vector.
+    ///                   If not required by the encryption algorithm, can be an empty octet sequence.
+    ///     * cipherText: The variable-size base64 encoded cipher text.
+    ///     * authTag: A fixed-size, 132-bit, base64 encoded authentication tag.
+    ///                Can be an empty octet sequence.
+    public func set(_ header: String, _ key: String, _ initVector: String, _ cipherText: String, _ authTag: String) {
+        guard !self.disabled else { return }
+
+        Dispatchers.shared.launchAPI {
+            glean_jwe_set(self.handle, header, key, initVector, cipherText, authTag)
+        }
+    }
+
+    /// Tests whether a value is stored for the metric for testing purposes only. This function will
+    /// attempt to await the last task (if any) writing to the the metric's storage engine before
+    /// returning a value.
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    /// - returns: true if metric value exists, otherwise false
+    public func testHasValue(_ pingName: String? = nil) -> Bool {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+        return glean_jwe_test_has_value(self.handle, pingName) != 0
+    }
+
+    // swiftlint:disable force_cast
+    /// Returns the stored value for testing purposes only. This function will attempt to await the
+    /// last task (if any) writing to the the metric's storage engine before returning a value.
+    ///
+    /// Throws a "Missing value" exception if no value is stored.
+    ///
+    /// - parameters:
+    ///     * pingName: represents the name of the ping to retrieve the metric for.
+    ///                 Defaults to the first value in `sendInPings`.
+    ///
+    /// - returns:  value of the stored metric
+    public func testGetValue(_ pingName: String? = nil) throws -> JweData {
+        Dispatchers.shared.assertInTestingMode()
+
+        let pingName = pingName ?? self.sendInPings[0]
+
+        if !testHasValue(pingName) {
+            throw "Missing value"
+        }
+
+        var data: JweData?
+
+        let jsonString = String(freeingRustString: glean_jwe_test_get_value_as_json_string(self.handle, pingName))
+        if let jsonData: Data = jsonString.data(using: .utf8, allowLossyConversion: false) {
+            if let json = try? JSONSerialization.jsonObject(with: jsonData, options: []) as? [String: Any] {
+                data = JweData(
+                    header: json["header"] as! String,
```

Comment:
Using the force cast (`as!`) could cause a crash here if anything went wrong here. Maybe it would be safer to provide a default value instead of crashing?   This can be done something like:
```suggestion
                    header: (json["header"] as? String) ?? "default",
```

Or you might also want to ensure that the value is available under that key before passing it into the `JweData` struct initializer.  If for some reason the value wasn't available for that key, it could also cause the force cast to fail/crash.

Reply:
If you say it's not possible to fail here, then I trust you :)

Reply:
I knew that this could cause a crash upon implementing, here is my thinking:
1. This is a testing only function;
2. A crash here would mean something is terribly wrong (this should never really crash, crashing would mean we either stored an invalid JWE value on `set` or that something went wrong while passing the JSON string through the FFI);

In this case I think crashing would be fine, since it is extremely unlikely and would be a helpful debugging crash.

Reply:
Okay, since this is test-only and it should never happen unless something is wrong, I am fine with using the force-cast here.